src/test/java/org/prebid/server/bidder/taboola/TaboolaBidderTest.java

```diff
@@ -0,0 +1,326 @@
+package org.prebid.server.bidder.taboola;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.iab.openrtb.request.Banner;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.request.Native;
+import com.iab.openrtb.request.Publisher;
+import com.iab.openrtb.request.Site;
+import com.iab.openrtb.request.Video;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.prebid.server.VertxTest;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.HttpResponse;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.taboola.ExtImpTaboola;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+import io.vertx.core.MultiMap;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+
+import static java.util.function.UnaryOperator.identity;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.assertj.core.api.Assertions.tuple;
+
+public class TaboolaBidderTest extends VertxTest {
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    private TaboolaBidder taboolaBidder;
+
+    @Before
+    public void setUp() {
+        taboolaBidder =
+                new TaboolaBidder("https://{{MediaType}}.bidder.taboola.com/OpenRTB/PS/auction/{{Host}}/{{PublisherID}}",
+                        "http://localhost-test.com",
+                        jacksonMapper);
+    }
+
+    @Test
+    public void createBidderWithWrongEndpointShouldThrowException() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new TaboolaBidder("incorrect.endpoint",
+                "http://localhost:8090",
+                jacksonMapper));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldTakeCurrencyAmountFromImp() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                impCustomizer -> impCustomizer.banner(Banner.builder().build()).bidfloorcur("USD"));
```

Comment:
No need to set and then assert `bidfloorcur`. Adapter doesn't modify it.
Same for other occurrences.